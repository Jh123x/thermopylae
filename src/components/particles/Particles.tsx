import { useEffect } from "react";
import { useSettings } from "../../hooks";

const Particles = () => {
    const { settings } = useSettings();
    useEffect(() => {
        window.particlesJS(
            "particles-js",
            settings.backgroundParticlesConfig,
            () => { }
        );
    });
    return (
        <div
            id="particles-js"
            style={{
                display: settings.backgroundParticlesEnabled ? "block" : "none",
            }}
        ></div>
    );
};

export default Particles;
