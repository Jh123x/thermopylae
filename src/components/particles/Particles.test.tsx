import { test, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import Particles from './Particles';

describe('Particles', () => {
    test('should render correctly', () => {
        const result = render(<Particles />);
        expect(result.baseElement).toMatchSnapshot();
    });
});