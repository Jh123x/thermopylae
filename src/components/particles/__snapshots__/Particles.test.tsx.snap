// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Particles should render correctly 1`] = `
<body>
  <div>
    <div
      id="particles-js"
      style="display: block;"
    />
  </div>
</body>
`;
